src/App.tsx

```diff
@@ -15,36 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+}
+
+interface TrackStatus {
+  reversed: boolean;
+  sortType: SortType;
+}
+
+const defaultState: TrackStatus = {
+  reversed: false,
+  sortType: SortType.Default,
+};
+
 export const App: React.FC = () => {
+  const [trackStatus, setTrackStatus] = useState<TrackStatus>(defaultState);
+
+  const { reversed, sortType } = trackStatus;
+
+  const getGoods = () => {
+    const goods = [...goodsFromServer];
+
+    if (sortType === SortType.Alphabetical) {
+      goods.sort((a, b) => a.localeCompare(b));
+    } else if (sortType === SortType.Length) {
+      goods.sort((a, b) => a.length - b.length);
+    }
+
+    if (reversed) {
+      goods.reverse();
+    }
+
+    return goods;
+  };
+
+  const handleSortAlphabetically = () => {
+    setTrackStatus({
+      ...trackStatus,
+      sortType: SortType.Alphabetical,
+      reversed: reversed,
+    });
+  };
+
+  const handleSortByLength = () => {
+    setTrackStatus({
+      ...trackStatus,
+      sortType: SortType.Length,
+      reversed: reversed,
+    });
+  };
+
+  const handleReverse = () => {
+    setTrackStatus({
+      ...trackStatus,
+      reversed: !reversed,
+    });
```

Comment:
It changes nothing in your case, but it is better to use the following:

```suggestion
    setTrackStatus(value => ({
      ...value,
      reversed: !value.reversed,
    }));
```